tools/ops/policystream.py

```diff
@@ -0,0 +1,610 @@
+        self.session = session
+        self.info = info
+
+    def send(self, change):
+        """send the given policy change"""
+        self.buf.append(change)
+        if len(self.buf) % self.BUF_SIZE == 0:
+            self.flush()
+
+    def flush(self):
+        """flush any buffered messages"""
+
+    def close(self):
+        self.flush()
+
+
+class KinesisTransport(Transport):
+
+    BUF_SIZE = 50
+
+    retry = staticmethod(get_retry(('ProvisionedThroughputExceededException',)))
+
+    def __init__(self, session, info):
+        self.session = session
+        self.info = info
+        self.client = self.session.client('kinesis', region_name=info['region'])
+        self.buf = []
+
+    def flush(self):
+        if not self.buf:
+            return
+        self.retry(
+            self.client.put_records,
+            StreamName=self.info['resource'],
+            Records=[
+                {'Data': json.dumps(c.data()),
+                 'PartitionKey': c.repo_uri}
+                for c in self.buf])
+        self.buf = []
+
+
+class SQSTransport(Transport):
+
+    BUF_SIZE = 10
+
+    def __init__(self, session, info):
+        self.session = session
+        self.info = info
+        self.client = self.session.client('sqs', region_name=info['region'])
+        self.buf = []
+
+    def flush(self):
+        if not self.buf:
+            return
+        self.client.send_message_batch(
+            QueueUrl=self.info['resource'],
+            Entries=[{
+                'Id': str(change.commit.id) + change.policy.name,
+                'MessageDeduplicationId': str(change.commit.id) + change.policy.name,
+                'MessageGroupId': change.repo_uri,
+                'MessageBody': json.dumps(change.data())}
+                for change in self.buf])
+
+
+class OutputTransport(Transport):
+
+    def send(self, change):
+        print(change)
+
+
+def transport(stream_uri, assume):
+    if stream_uri == 'stdout':
+        return OutputTransport(None, None)
+    if not stream_uri.startswith('arn'):
+        raise ValueError("invalid transport")
+    info = parse_arn(stream_uri)
+    session_factory = boto3.Session
+    if assume:
+        session_factory = partial(SessionFactory, assume_role=assume)
+    if info['service'] == 'kinesis':
+        return KinesisTransport(session_factory(), info)
+    elif info['service'] == 'sqs':
+        qurl = "https://sqs.{region}.amazonaws.com/{account}/{resource}".format(**info)
+        info['resource'] = qurl
+        return SQSTransport(session_factory(), info)
+
+    raise ValueError("unsupported transport %s" % stream_uri)
+
+
+@click.group()
+def cli():
+    """Policy changes from git history"""
+
+
+@cli.command(name='diff')
+@click.option('-r', '--repo-uri')
+@click.option('--source', default='master', help="source/baseline revision spec")
+@click.option('--target', default=None, help="target revisiion spec")
+@click.option('-o', '--output', type=click.File('wb'), default='-')
```

Comment:
I prefer -s here as opposed to -o as it falls more in line with the rest of c7n's cli syntax

Reply:
-s in that context references a directory, this is just an output file...

Reply:
there's also '-s' for stream-uri in the stream command already, not sure its worthwhile outside on the consistency with a separate tool, there's not really a standard here.